camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java

```diff
@@ -476,9 +478,23 @@ public void sendTrackingCounts(JobContext job, FileSystem fs, Path newExecutionO
         }
       }
 
+      int publishCountMaxRetries =
+          Integer.parseInt(props.getProperty(ETL_PUBLISH_COUNT_MAX_RETRIES, ETL_PUBLISH_COUNT_MAX_RETRIES_DEFAULT));
       String brokerList = getKafkaBrokers(job);
       for (EtlCounts finalCounts : allCounts.values()) {
-        finalCounts.postTrackingCountToKafka(job.getConfiguration(), props.getProperty(KAFKA_MONITOR_TIER), brokerList);
+        for (int i = 0; i <= publishCountMaxRetries; i++) {
+          try {
+            finalCounts.postTrackingCountToKafka(job.getConfiguration(), props.getProperty(KAFKA_MONITOR_TIER),
```

Comment:
I think we need to put the retry inside the kafka publishing, retry on the high level loop would cause lots of duplicate publishing.